File: 5.longest-palindromic-substring.cs
/*
 * @lc app=leetcode id=5 lang=csharp
 *
 * [5] Longest Palindromic Substring
 */

// @lc code=start
public class Solution {
    public int check(string s, int i, int j){
        int m = (j-i)/2;
        int bb = i;
        for(int k=i; k<m+bb+1; k++){
            if(s[k]!=s[j]){
                return 0;
            }
            j--;
        }
        return 1;
    }
    public string LongestPalindrome(string s) {
        int n = s.Length;
        int w = n;
        for(int i=n-1; i>=0; i--){
            int b = 0;
            int e = i;
            for(int k = i; k<n; k++){
                string sub = s.Substring(b, e+1);
                if(check(s, b, k)==1){
                    return sub;
                }
                b++;
            }
        }
        string res = "";
        return res+s[0];
    }
}
// @lc code=end

File: 5.longest-palindromic-substring.cpp
/*
 * @lc app=leetcode id=5 lang=cpp
 *
 * [5] Longest Palindromic Substring
 */

// @lc code=start
class Solution {
public:
    string longestPalindrome(string s) {
        int n = s.size();
        int maxlen =1;
        int start=0;
        int dp[1001][1001] = {0};
        for(int j=0; j<n; j++){
            for(int i=0; i<j; i++){
                if(s[i]==s[j]&&(j-i<=2||dp[i+1][j-1]==1)){
                    dp[i][j] = 1;
                    if(j-i+1>maxlen){
                        maxlen = j-i+1;
                        start = i;
                    }
                    
                }
            }
        }
        return s.substr(start, maxlen);
    }
};
// @lc code=end

